### ISSUE TITLE
Equation numbering for "gather" is not working.

### ISSUE THREAD

**Issue by berkorbay** [quarto-dev/quarto-cli#2870]
### Bug description

When I used RMarkdown following tex code used to yield equation numbers and I could get references. But in Quarto it seems to be no longer possible. My .qmd file is part of a Quarto Book. Neither preview nor render made a difference.

> \begin{gather}
> \max z = 3x_{1} + 2x_{2} \label{eq:obj.fun} \\
> s.t. \nonumber \\
> 2x_{1} + x_{2} \le 100 \label{eq:finishing.constraint} \\
> x_{1} + x_{2} \le 80 \label{eq:carpentry.constraint}\\
> x_{1} \le 40 \label{eq:demand.constraint}\\
> x_{1},x_{2} \ge 0 \\
> \end{gather}
> 
> See \ref{eq:obj.fun}

I saw this [cross references](https://quarto.org/docs/authoring/cross-references.html#equations) section on Quarto but having too many dollar signs is hard on readability. Is there a way to get numbering back?

My RStudio version is 2022.07.1 Build 554 on MacOS Monterey 12.2.1

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
I'm not sure I follow this:

> I saw this [cross references](https://quarto.org/docs/authoring/cross-references.html#equations) section on Quarto but having too many dollar signs is hard on readability.

Can you clarify? Also, can you share a full reproducible example we can test locally? Thank you.

**Comment by mcanouil** [comment]
This seems to be more of a MathJax (is it MathJax?) configuration issue.
Maybe of interest, few weeks ago, there was a question to number and reference equation in Quarto but not as usual/default because the rendering differed between LaTeX and HTML.

````
---
title: "Untitled"
format:
  html:
    html-math-method: mathjax
    include-in-header:
      - text: |
          <script>
          window.MathJax = {
            tex: {
              tags: 'ams'
            }
          };
          </script>
  pdf: default
---

This is a multiline equation using the `split` environment.

\begin{align}
  y &= ax + b \notag \\
    &= cx + d \notag \\
    &= \ldots = 4 \label{eq-test}
\end{align}

And I can reference the last line via \ref{eq-test}
````

[^1]: <https://twitter.com/MickaelCanouil/status/1574718294229061633>

**Comment by cscheid** [comment]
I'm glad you folks found that workaround. I'm not sure I want to make that the "official" solution because I imagine it won't work predictably across formats.

**Comment by mcanouil** [comment]
The issue is that MathJax and other tools like this does not have the exact same behaviour as LaTeX.
Here, we had to enable a MathJax option to ensure HTML rendering was like the one in LaTeX.

This is also because Pandoc does not require `$$` which is also one source of discrepancy between HTML and LaTeX.

**Comment by berkorbay** [comment]
Hi. Thank you for all your answers. 

1. First of all, I have to say I'm sorry. There was a confusion on my end. I'm used to use `gather` rather than equation, align and other beautiful LaTeX equation tools. In the previous months I usually rendered Rmd files to PDF files and gather had equation numbers and referencing. It worked quite nice. When I switched to Quarto, I expected similar behavior. Though I could not get equation numbering or references (hence ???) stuff. Since Quarto is quite new to me, I thought this might be a bug. What I missed to notice was this time I was rendering to HTML 🤦‍♀️. 
 
2. Thanks to @mcanouil I now have a way to implement equation numbering and referencing functionality 🎉

3. @cscheid What I meant by cross referencing section is that the only documentation which I found remotely relevant  was that one. It used dollar signs (as expected from HTML rendering) but I did not want to use those since I also want to keep my qmd file as lean as possible. My first confusion (pdf-html latex differences) might also have caused confusion on this one unintentionally, sorry about that.

4. Quarto works as intended, therefore I'm closing my bug report. As a side note, I switched to Quarto quite conveniently. I also started teaching Quarto this semester! Student work is displayed as Quarto Book on their GH pages. Thank you all for building such a nice software🎉

5. Here is an MWE just to complete the issue. 

```
---
title: "MWE Gather"
format: html
editor: visual
---
Here is a minimal working example on gather.

\begin{gather}
\max z = 3x_{1} + 2x_{2} \label{eq:obj.fun} \\
s.t. \nonumber \\
2x_{1} + x_{2} \le 100 \label{eq:finishing.constraint} \\
x_{1} + x_{2} \le 80 \label{eq:carpentry.constraint}\\
x_{1} \le 40 \label{eq:demand.constraint}\\
x_{1},x_{2} \ge 0 \\
\end{gather}

See \ref{eq:obj.fun}
```
